org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodAnalyzer.java

```diff
@@ -103,10 +107,17 @@
 	public MethodAnalyzer(final String className, final String superClassName,
 			final String name, final String desc, final String signature,
 			final boolean[] probes) {
+		this(className, superClassName, name, desc, signature, probes, FILTERS);
+	}
+
+	MethodAnalyzer(final String className, final String superClassName,
```

Comment:
Now as we have a instance member with the filters I don't think this class is the right set to maintain the list of all filters.

I would prefer to move the `FILTERS` constant to a new class `AllFilters` in the filter package.

Also there is no need to have a filter array: We can have a aggregate Filter which combines a list of filters into one `IFilter` instance. This simplifies the code in this class.

Reply:
> Also there is no need to have a filter array

Presence of array simplifies disabling of filters in `MethodAnalyzerTest`, otherwise will also need `NopFilter`.

Reply:
@marchof extracted, let's see if can pass review now :laughing: